### DISCUSSION TITLE
peer recognition of tunnel down.

### DISCUSSION THREAD

**Discussion by hcojo** [strongswan/strongswan#965] [Q&A] [answered]
hi, 
I encounter an issue on the next scenario: 
tunnel is up between two hosts, and traffic is transferred in the tunnel due to _http get_ of a large file. 
during traffic, on the client side, i perform 'ip xfrm state flush'

i would expect that the server side will identify the tunnel is down, and will try to re-establish a new one. 

also - when trying to manually ping from server side to client side, it failed.
only when pinging from client side to server - the tunnel was reestablished. 

this is the configuration i use: 
----------------------------------
connections {
    pok1-qz1-sr1-rk003-s17-10-66-3-35-ipv4 {
        local_addrs = pok1-qz1-sr1-rk025-s17.ipv4.vcn.vpc.cloud.ibm.local
        remote_addrs = pok1-qz1-sr1-rk003-s17.ipv4.vcn.vpc.cloud.ibm.local
        dpd_delay = 0
        local {
            auth = psk
            # no cert required for psk
        }
        remote {
            auth = psk
        }
        children {
            pok1-qz1-sr1-rk003-s17-10-66-3-35-ipv4 {
                local_ts = 11.6.81.0/24
                remote_ts = 11.0.209.0/24
                mode = tunnel
                inactivity = 130
                start_action = trap
                dpd_action = clear
                esp_proposals = aes256gcm8-sha256-modp2048
                rekey_time=60m
                rand_time=6m
                interface=ipsec0
                mark_in=5
            }
        }
        version = 2
        proposals = aes256gcm8-sha256-modp2048
        rekey_time=240m
        rand_time=24m
    }
}

**Comment by strongX509** [comment, marked as answer]
If DPD (Dead Peer Detection) is not enabled on the server it will never know that the IKE_SA was destroyed due to the external xfrm flushing on the client. Even the client will not get any message via XFRM when the IKE_SA is flushed. But since the IPsec policy is trap, a ping on the client side will re-negotiate the IPsec_SA. With dpd_delay = 0, DPD is disabled on the client anyway but in the case of xfrm flushing DPD is not needed since an outbound packet will re-establish the IKE_SA immediately.

**Reply by hcojo** [reply]
hi @strongX509 
thanks for the fast reply. 

i changed the dpd_delay to 2s (on both server and client hosts) but it didn't help for flush case.

just to be on the safe side, i want to emphasize that i'm looking for a solution where: 
in the middle of file download, i "state flush" in the client side - and i want to download to continue without triggering any other packets.

**Reply by tobiasbrunner** [reply]
> that the IKE_SA was destroyed due to the external xfrm flushing on the client

That does not actually destroy the IKE_SA, neither peer will notice this and DPD doesn't change this as the IKE_SA will still be responsive. Until the CHILD_SA is reestablished (on the client due to the trap policy, the server can't recreate it), no traffic can be transported.

> i "state flush" in the client side

Why should that happen?

**Reply by hcojo** [reply]
@tobiasbrunner 
i guess it should not happen.
just using it for some testing.
